paddle/gserver/layers/DetectionOutputLayer.cpp

```diff
@@ -139,7 +139,13 @@ void DetectionOutputLayer::forward(PassType passType) {
                                        allDecodedBBoxes,
                                        &allIndices);
 
-  resetOutput(numKept, 7);
+  if (numKept > 0) {
+    resetOutput(numKept, 7);
+  } else {
+    MatrixPtr outV = getOutputValue();
+    outV = NULL;
```

Comment:
Better to add comments in the Python API. It is useful for users to know the output can be null if there are no detection results.